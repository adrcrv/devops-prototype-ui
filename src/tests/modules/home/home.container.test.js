import React from 'react';
import { render } from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import HomeContainer from '../../../app/modules/home/home.container';

beforeEach(() => {
  jest.clearAllMocks();
});

describe('Home Container', () => {
  it('Expect Home to render && uptime to be 0H 0M 5S', () => {
    process.env.NEXT_PUBLIC_LAST_UPDATE = 'Sat Apr 23 02:13:45 -03 2022';
    jest.useFakeTimers().setSystemTime(new Date(process.env.NEXT_PUBLIC_LAST_UPDATE));
    const { getByText, rerender } = render(<HomeContainer />);
    act(() => {
      jest.advanceTimersByTime(5000);
      rerender(<HomeContainer />);
    });
    expect(getByText('0H 0M 5S')).toBeInTheDocument();
  });

  it('Expect Home to render unchanged', () => {
    process.env.NEXT_PUBLIC_LAST_UPDATE = 'Sat Apr 23 02:13:45 -03 2022';
    jest.useFakeTimers().setSystemTime(new Date(process.env.NEXT_PUBLIC_LAST_UPDATE));
    const { container } = render(<HomeContainer />);
    expect(container).toMatchSnapshot();
  });
});
