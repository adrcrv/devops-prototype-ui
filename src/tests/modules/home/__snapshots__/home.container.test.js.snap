// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Container Expect Home to render unchanged 1`] = `
<div>
  <div>
    <div
      class="faceWrapper"
    >
      <div>
        <div
          style="text-align: center;"
        >
          <h2>
            Welcome to DevOps TCC
          </h2>
          <hr
            width="150px"
          />
        </div>
        <div
          class="faceCardWrapper"
        >
          <div
            class="faceCard"
          >
            <h4>
              Last Deploy
            </h4>
            <p>
              23/04/2022 02:13:45
            </p>
          </div>
          <div
            class="faceCard"
          >
            <h4>
              Uptime
            </h4>
            <p>
              Loading...
            </p>
          </div>
          <div
            class="faceCard"
          >
            <h4>
              Random Content
            </h4>
            <p
              style="font-size: 24px;"
            >
              ðŸ¤–
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="shockingPhraseWrapper"
    >
      <div>
        <h2>
          â€œDevOps does deal with automation,
          <br />
          Â Â but that's not its only function.â€œ
        </h2>
      </div>
      <div>
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27400%27%20height=%27200%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="robot"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
    </div>
  </div>
</div>
`;
